### DISCUSSION TITLE
Please consider making `AuditEvent.Environment.Exception` return the actual Exception

### DISCUSSION THREAD

**Discussion by rlgordey** [thepirat000/Audit.NET#447] [Ideas]
Please consider making `AuditEvent.Environment.Exception` return the actual Exception or have a `.AsException` extension method.

Awesome project!

**Comment by thepirat000** [comment]
As far as I know, there is no way to do that. 

This is the code where we get the exception from interop
https://github.com/thepirat000/Audit.NET/blob/8dd2499b7b955a97c06a3ba36d36f95d682cf500/src/Audit.NET/AuditScope.cs#L393-L407

The `Environment.Exception` property should only be used to determine if there is an exception or not, but to get the real exception the only way is to add a try/catch block, as some of the extension libraries does, ([EF](https://github.com/thepirat000/Audit.NET/blob/8dd2499b7b955a97c06a3ba36d36f95d682cf500/src/Audit.EntityFramework/DbContextHelper.cs#L479), [WebApi](https://github.com/thepirat000/Audit.NET/blob/8dd2499b7b955a97c06a3ba36d36f95d682cf500/src/Audit.WebApi/AuditMiddleware.cs#L74), etc)

Please check the following SO questions:
- https://stackoverflow.com/questions/65981155/alternative-for-marshal-getexceptioncode
- https://stackoverflow.com/questions/3551873/how-to-detect-if-a-program-is-executing-under-a-thrown-exception-at-runtime

**Reply by rlgordey** [reply]
@thepirat000 Thanks for your consideration.
